### PR TITLE
feat: add commitlint

### LINKED ISSUE #267: [DOC-295] Add commitlint for linting of commit messages
As the project grows we need to work on enforcing consistency across various processes, one of these areas is commit messages and formatting.

To handle this we would like to use [commitlint](https://github.com/conventional-changelog/commitlint) in combination with [husky](https://github.com/typicode/husky) (which is already in the repo). Using commitlint we would like to enforce [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) which will require that commits be labelled based on the kind of change they are.

This change will help align the way we contribute to the project and also allow niceties with other tools for automated release notes and versioning.

Implementation of this change should be fairly straightforward, from memory there is a quick start that will assist with the implementation of most of this without any real trouble.

<sub>From [SyncLinear.com](https://synclinear.com) | [DOC-295](https://linear.app/documenso/issue/DOC-295/add-commitlint-for-linting-of-commit-messages)</sub>

### PR DESCRIPTION
**Pull request by Ashutosh-Bhadauriya**
Adds commit lint

Fixes #267
/claim #267

### CODE REVIEW

**Comment by Mythie** [comment]
Hey there, it looks like you haven't accepted our contributor license agreement yet. In order for us to accept your pull request we ask that you please fill out the CLA:

https://documen.so/cla

**Comment by Ashutosh-Bhadauriya** [comment]
Just did it, I thought I already did that 😅

File: .husky/commit-msg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npx --no -- commitlint --edit $1
```
Comment (Mythie) [inline .husky/commit-msg:4]: Can we make this an npm script instead of `npx`, that way it can be a first class citizen 😄 

e.g. `npm run commitlint $1`

**Comment by Ashutosh-Bhadauriya** [comment]
Just did it 😄

**Comment by Mythie** [comment]
Sweet, last but not least can we get a cool little demo video 👀 🚀

**Comment by Ashutosh-Bhadauriya** [comment]
Here we go 🚀

https://github.com/documenso/documenso/assets/62984427/6ab4b87b-ec85-4afc-a26c-07f033afe298

**Review by Mythie — APPROVED** [review]

**Comment by Mythie** [comment]
Thanks Ash!

thanks-guitar

**Pull request merged**